source_middleware.go

```diff
@@ -690,6 +690,17 @@ func (s *sourceWithBatch) Open(ctx context.Context, pos opencdc.Position) error
 	s.readCh = make(chan readResponse, *s.config.BatchSize)
 	s.readNCh = make(chan readNResponse, 1)
 
+	if *s.config.BatchSize > 0 && *s.config.BatchDelay == 0 {
+		Logger(ctx).Warn().
+			Msg("batch size is set but batch delay is not, this might result in the connector waiting indefinitely for the batch to be filled")
```

Comment:
I would use the actual configuration options to make it explicit

```suggestion
			Msg("sdk.batch.size is set but sdk.batch.delay is not, this might result in the connector waiting indefinitely for the batch to be filled")
```